src/components/Popup/Popup.js

```diff
@@ -25,7 +25,10 @@ const Popup = () => {
 
   const { coordinate, features } = clickedFeatureInfo[0];
   const geom = features[0].getGeometry();
-  const coord = geom instanceof Point ? geom.getCoordinates() : coordinate;
+  const coord =
+    features.length === 1 && geom instanceof Point
+      ? geom.getCoordinates()
+      : coordinate;
   const mapRect = map.getTarget().getBoundingClientRect();
```

Comment:
I've also changed the core popup:
* This shows the popup at the click position when multiple features are clicked. Previously, the coordinate of the first feature was used, although the popup position does not update when going to the next feature in the popup
* And this allows to set a custom coordinate for cluster features.